### PR TITLE
Improve handling of bytecode files

### PR DESCRIPTION
**Pull request by mdboom**
This improves a few things about handing compiled bytecode files (`.pyc` files inside of `__pycache__` directories).

It was measured quite some time ago that it is faster to just recompile Python source files to bytecode files every time than to transfer them and load them from the virtual filesystem (since the source file always has to be loaded anyway).  And if one does that, there's no need to write the bytecode files to the virtual filesystem since they are ephemeral and will disappear on page reload anyway.

Unfortunately, there were a few things that have accidentally broken around this along the way:

- Packages were including `__pycache__` directories due to an incorrect `exclude` pattern passed to `file_packager.py`.  This should decrease filesizes -- for example matplotlib goes from 15.8 MB to 12.9MB
- Setting the `PYTHONDONTWRITEBYTECODE` environment variable actually has no effect.  I think it's only read by CPython as part of the commandline argument parsing step, which we don't do.  Instead, this now forcibly sets `sys.dont_write_bytecode`, which does seem to work.
- Setting `sys.implementation.cache_tag` to `None` prevents the module loader from looking for `.pyc` files, which saves some additional "virtual disk I/O"

With these changes `import matplotlib.pyplot` goes from 7.1s to 5.9s in Firefox.

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -41,11 +41,40 @@ main(int argc, char** argv)
...
+  // This doesn't seem to work anymore, but I'm keeping it for good measure anyway
+  // The effective way to turn this off is below: setting sys.done_write_bytecode = True
+  setenv("PYTHONDONTWRITEBYTECODE", "1", 0);
+
+  PyObject* sys = PyImport_ImportModule("sys");
+  if (sys == NULL) {
+    return 1;
+  }
+  if (PyObject_SetAttrString(sys, "dont_write_bytecode", Py_True)) {
+    Py_DECREF(sys);
```
Comment (wlach) [inline src/main.c]: Is there any reason to decrement the reference count if we're just going to return 1 anyways?

Comment (mdboom) [reply src/main.c]: Probably not from the main function here, no.  Good point.

**Comment by mdboom** [comment]
So, the change to not look for `.pyc` files is breaking a number of tests, but only because the tests are written to clear cache files for testing purposes, not because the tests themselves indicate anything broken.  I think I'll back that part of this out for now and reinvestigate later.

**Comment by rth** [comment]
This is a very nice improvement for the package sizes!

**Pull request merged**